### ISSUE TITLE
fec: cc enc/dec in ber_curve_gen example yields different results in 3.8

### ISSUE THREAD

**Issue by mormj** [gnuradio/gnuradio#2642]
When running the ber_curve_gen.grc example in gr-fec, the convolutional coder results do not look right at all as compared to the graph produced in 3.7

Here is the graph produced under 3.8:
![image](https://user-images.githubusercontent.com/34754695/61540217-3742d600-aa0b-11e9-9ebc-4ec7f29fdbb4.png)

Under 3.7
![image](https://user-images.githubusercontent.com/34754695/61540258-49247900-aa0b-11e9-8079-65176bc54c85.png)

The parameters for the objects in the .grc file all look the same between both

**Comment by jdemel** [comment]
It looks like the CC code fails. Besides, the curves seem to be correct. If the simulation would run for a couple more hours, I'd guess, those curves would look the same. I suggest to investigate, why the CC code fails. What about CC code tests? Do they pass?

**Comment by fmagno** [comment]
There seems to be an issue with the example `fec/examples/fecapi_cc_decoders.grc` that could be related:
![image](https://user-images.githubusercontent.com/3079603/73967465-dda2ea00-490f-11ea-8d1f-8c5b2b8ff824.png)

**Comment by marcusmueller** [comment]
Currently broken, can't instantiate Bercurve Sink:

```
Param - Bus_Structure_Sink(bus_structure_sink):
	Value "list(map(lambda b:list(map(lambda a:b * len(esno) * 2 + a, range(len(esno)*2))), range(num_curves)))" cannot be evaluated:
	name 'num_curves' is not defined
```

although `num_curves` is a parameter

**Comment by marcusmueller** [comment]
Depends on #7044

**Comment by argilo** [comment]
I expect this was due to the broken AVX2 convolutional decoder in VOLK, which was removed in https://github.com/gnuradio/volk/pull/458. I recently tracked down the bugs in https://github.com/gnuradio/volk/pull/736#issuecomment-1868383520.

**Comment by argilo** [comment]
To get the BER curve example working again, I applied the workaround described in https://github.com/gnuradio/gnuradio/issues/5627#issuecomment-1060608139, and reverted #6889 to solve #7059.

![Screenshot from 2024-01-06 12-02-07](https://github.com/gnuradio/gnuradio/assets/583749/99321e81-3808-47f0-bb97-3f0982cbf4dc)

The convolutional decoder seems to work fine now, but LDPC appears broken. :cry:

**Comment by argilo** [comment]
If I revert #6748, then LDPC appears to work again:

![Screenshot from 2024-01-06 12-35-00](https://github.com/gnuradio/gnuradio/assets/583749/9f473369-1e2a-4778-ac10-2230fd662260)

**Comment by argilo** [comment]
I'll close this off, since the original bug with the convolutional decoder was fixed in https://github.com/gnuradio/volk/pull/458.

I opened #7060 to cover the new issue with the LDPC decoder.
